File: src/main/java/kitchenpos/menu/tobe/domain/MenuPrice.java
package kitchenpos.menu.tobe.domain.vo;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import kitchenpos.exception.IllegalPriceException;

@Embeddable
public class MenuPrice {
    @Column(name = "price", nullable = false)
    private long price;

    protected MenuPrice() {
    }

    public MenuPrice(Long price) {
        validatePrice(price);
        this.price = price;
    }

    private static void validatePrice(Long price) {
        if (price < 0) {
            throw new IllegalPriceException("가격은 0원 미만일 수 없습니다. ", price);
        }
    }

    public Long getPrice() {
        return price;
    }
}

File: src/main/kotlin/kitchenpos/menu/tobe/domain/MenuPrice.kt
package kitchenpos.menu.tobe.domain

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import java.math.BigDecimal

@Embeddable
class MenuPrice(
    @Column(name = "price", nullable = false)
    val price: BigDecimal
) {

    init {
        require(price >= BigDecimal.ZERO) { "메뉴 가격은 0 이상이어야 합니다." }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as MenuPrice

        return price == other.price
    }

    override fun hashCode(): Int {
        return price.hashCode()
    }
}
